src/Chemistry.hpp

```diff
@@ -142,7 +158,19 @@ template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const R
 				state(i, j, k, HydroSystem<problem_t>::scalar0_index + nn) = inmfracs[nn] * rho; // scale by rho to return partial densities
 			}
 		});
+
+#if defined(AMREX_USE_HIP)
+		amrex::Gpu::streamSynchronize(); // otherwise HIP may fail to allocate the necessary resources.
+#endif
 	}
+
+#if defined(AMREX_USE_GPU)
+	num_failed = *(d_num_failed.copyToHost());
+#endif
+
+	burn_success = !num_failed;
+
+	amrex::ParallelDescriptor::ReduceIntMin(burn_success);
```

Comment:
You have to do something here with the value of `burn_success` at the end of this function. The reasonable options are to either i) abort the code if burn_success is false, or ii) return the value of burn_success, and let the hydro update figure out what to do.